core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -24,27 +24,32 @@ import com.fasterxml.jackson.databind.annotation.JsonDeserialize
 
 import org.apache.spark.JobExecutionStatus
 
-class ApplicationInfo private[spark](
-    val id: String,
-    val name: String,
-    val coresGranted: Option[Int],
-    val maxCores: Option[Int],
-    val coresPerExecutor: Option[Int],
-    val memoryPerExecutorMB: Option[Int],
-    val attempts: Seq[ApplicationAttemptInfo])
+case class ApplicationInfo private[spark](
+    id: String,
+    name: String,
+    coresGranted: Option[Int],
+    maxCores: Option[Int],
+    coresPerExecutor: Option[Int],
+    memoryPerExecutorMB: Option[Int],
+    attempts: Seq[ApplicationAttemptInfo]) {
+
+    def completed: Boolean = {
```

Comment:
Is this used anywhere?

Reply:
org.apache.spark.deploy.history.HistoryPage:33

Reply:
Can we move this logic there instead? That avoids adding a new visible property in a public object.

Reply:
Moved, also had to add it to `HistoryServerSuite`.
A bit of code duplication, but indeed better than increasing API surface area.